tfe/data_source_oauth_client_test.go

```diff
@@ -3,32 +3,65 @@ package tfe
 import (
 	"fmt"
 	"math/rand"
+	"regexp"
 	"testing"
 	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-func TestAccTFEOAuthClientDataSource_basic(t *testing.T) {
+func testAccTFEOAuthClientDataSourcePreCheck(t *testing.T) {
+	testAccPreCheck(t)
+	if GITHUB_TOKEN == "" {
+		t.Skip("Please set GITHUB_TOKEN to run this test")
+	}
```

Comment:
I started to do this, but tripped over a build error when adding it to `tfe/testing.go` file with the other `skipIf` functions. It looks like `testing.go` is not getting excluded from non-test builds because of its name, but referencing the `GITHUB_TOKEN` global var (defined in `provider_test.go`) is thus not allowed.

Looks like `testing.go` probably ought to be renamed, but that probably needs to happen in a different PR since that might impact other things if there's any other leakage between that file and non-test code.